lib/game.rb

```diff
@@ -1,50 +1,104 @@
-require_relative "console/user_interaction"
-require_relative "language"
-require_relative "board"
-require_relative "tictactoe"
-require_relative "human"
-require_relative "computer"
-
+require_relative 'console/user_interaction'
+require_relative 'language'
+require_relative 'board'
+require_relative 'tictactoe'
+require_relative 'human'
+require_relative 'computer'
+# This class handles the set up of the game
 class Game
-
   def start
+    ask_to_play
+    create_players
+    board = Board.new(@ui, @markers)
+    tictactoe = TicTacToe.new(@players, board, @ui)
+    tictactoe.set_current_player(@hum, @com, @player1, @player2)
+    @current_player = tictactoe.current_player
+    @ui.board
+    board.print_board
+    @ui.first(@current_player.name)
+    tictactoe.play(board)
+  end
+
+  def create_players
+    @markers = Marker.new
+    mode = set_mode
+    @name = set_user_name
+    define_difficulty_level(@name) unless @ui.human(mode)
+    human_vs_computer if @ui.computer_human(mode)
+    human_vs_human if @ui.human(mode)
+    computer_vs_computer if @ui.computer(mode)
```

Comment:
I'm thinking maybe we can make this a bit cleaner by changing how the UI works. Instead of asking the user which game mode it is, you can ask:

* Is player 1 computer or human?
* Is player 2 computer or human?

These two questions should cover all the possibilities of human vs human, human vs computer, computer vs human, computer vs computer.

Reply:
implemented in 2d6b918